go/libkb/globals.go

```diff
@@ -966,12 +966,12 @@ func (g *GlobalContext) ClearSharedDHKeyring() {
 
 // BumpSharedDHKeyring recreates SharedDHKeyring if the uid/did changes.
 func (g *GlobalContext) BumpSharedDHKeyring() error {
-	g.sharedDHKeyringMu.Lock()
-	defer g.sharedDHKeyringMu.Unlock()
-
 	uid := g.GetMyUID()
 	did := g.Env.GetDeviceID()
 
+	g.sharedDHKeyringMu.Lock()
+	defer g.sharedDHKeyringMu.Unlock()
```

Comment:
Maybe include some comments about how important it is not to call `GetMyUID` inside of this lock?